### PR TITLE
#187 - holiday day counter

### PR DESCRIPTION
**Pull request by kamilpiech97**
Added information how many days left to holiday day. If holiday is current day - information should not be shown.
Preview:
image

### CODE REVIEW

File: app/Infrastructure/Http/Resources/HolidayResource.php [hunk 1]
```diff
@@ -19,6 +20,7 @@ public function toArray($request): array
             "isPast" => $this->date->endOfDay()->isPast(),
             "displayDate" => $this->date->toDisplayString(),
             "dayOfWeek" => $this->date->dayName,
+            "daysToHoliday" => $this->date->diffInDays(Carbon::now()),
```
Comment (krzysztofrewak) [inline app/Infrastructure/Http/Resources/HolidayResource.php:23]: Maybe `->diffForHumans()` would be better here?

File: resources/js/Shared/Widgets/UpcomingHolidays.vue [hunk 2]
```diff
@@ -18,6 +18,9 @@
             </p>
             <p class="text-sm text-gray-500">
               {{ holiday.displayDate }} ({{ holiday.dayOfWeek }})
+              <span v-if="holiday.daysToHoliday !== 0">
+                - <strong>{{ holiday.daysToHoliday }}</strong>
```
Comment (krzysztofrewak) [inline resources/js/Shared/Widgets/UpcomingHolidays.vue:22]:
```suggestion
                - {{ holiday.daysToHoliday }}
```

I would drop this bolding.

File: app/Infrastructure/Http/Resources/HolidayResource.php [hunk 3]
```diff
@@ -19,6 +19,10 @@ public function toArray($request): array
             "isPast" => $this->date->endOfDay()->isPast(),
             "displayDate" => $this->date->toDisplayString(),
             "dayOfWeek" => $this->date->dayName,
+            "daysToHoliday" => $this->date->diffForHumans([
+                "parts" => 2,
+                "join" => " i ",
+            ]),
```
Comment (krzysztofrewak) [inline app/Infrastructure/Http/Resources/HolidayResource.php:25]: Why not simpler?
```suggestion
            "daysToHoliday" => $this->date->diffForHumans(),
```

Comment (kamilpiech97) [reply app/Infrastructure/Http/Resources/HolidayResource.php:25]: it is show only weeks or month when holiday will became but its not problem to change that.

Comment (krzysztofrewak) [reply app/Infrastructure/Http/Resources/HolidayResource.php]: I think that "in 2 months" in more readable than "in 2 months and 12 days". ;)

**Review by EwelinaLasowy — APPROVED** [review]

**Pull request merged**